src/knowledge_commitment/knowledge_commitment.hpp

```diff
@@ -60,8 +60,8 @@ struct knowledge_commitment {
 template<typename T1, typename T2, mp_size_t m>
 knowledge_commitment<T1,T2> operator*(const libff::bigint<m> &lhs, const knowledge_commitment<T1,T2> &rhs);
 
-template<typename T1, typename T2, mp_size_t m, const libff::bigint<m> &modulus_p>
-knowledge_commitment<T1,T2> operator*(const libff::Fp_model<m, modulus_p> &lhs, const knowledge_commitment<T1,T2> &rhs);
+template<typename T1, typename T2, typename T>
```

Comment:
Agree, this is what caused me to feel uneasiness as well. 
As far as it seems to me, it's a different interpretation regarding non-type template parameter deduction - gcc decides to infer "bigint<n>" from "const bigint<n> &" while clang is being more strict and decides not to.

This is a contained example that shows the same behavior:
http://coliru.stacked-crooked.com/a/583904c911c2e84b